subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaExecToolchainIntegrationTest.groovy

```diff
@@ -14,17 +14,17 @@
  * limitations under the License.
  */
 
-package org.gradle.api.tasks;
+package org.gradle.api.tasks
+
 
-import org.gradle.api.JavaVersion
 import org.gradle.integtests.fixtures.AbstractPluginIntegrationTest
 import org.gradle.integtests.fixtures.AvailableJavaHomes
 import org.gradle.internal.jvm.Jvm
 import spock.lang.IgnoreIf
 
 class JavaExecToolchainIntegrationTest extends AbstractPluginIntegrationTest {
 
-    @IgnoreIf({ AvailableJavaHomes.differentJdk == null })
+    @IgnoreIf({ AvailableJavaHomes.differentVersion == null })
```

Comment:
These tests have turned nondeterministic precisely because of what you are saying in your other comment, ie. because I made it pick the current JVM even when auto-detection wasn't enabled. Will revert.